extras/task_yaml_path_checker.py

```diff
@@ -0,0 +1,93 @@
+from typing import Dict
+
+import neurobooth_os.iout.metadator as meta
+from neurobooth_os.iout.stim_param_reader import StudyArgs, CollectionArgs, RawTaskParams, InstructionArgs, \
+    StimulusArgs, DeviceArgs, SensorArgs
+
+
+class PathError(Exception):
+    pass
+
+
+def check_paths_from_study(study_name: str):
+    """
+    Reads and parses all yaml files found in NB_CONFIG folder. The named study is checked to
+    ensure that every element referenced from that study has an associated yaml file.
+
+    Parameters
+    ----------
+    study_name Name of the study to check
+
+    Returns
+    -------
+
+    """
+    print("Note: Checking against yaml files in location specified in the NB_CONFIG environment variable.")
+    studies: Dict[str, StudyArgs] = meta.read_studies()
+    if study_name not in studies:
+        raise PathError(f"No yaml file was found for study '{study_name}' ")
+    else:
+        study = studies[study_name]
+        collections: Dict[str, CollectionArgs] = meta.read_collections()
+        collection_ids = study.collection_ids
+        for collection_id in collection_ids:
+            if collection_id not in collections:
+                raise PathError(
+                    f"Collection '{collection_id}' referenced by study '{study}' was not found in yaml files")
+            else:  # check the collection
+                task_ids = collections[collection_id].task_ids
+                tasks: Dict[str, RawTaskParams] = meta.read_tasks()
+                for task_id in task_ids:
+                    if task_id not in tasks:
+                        raise PathError(f"Task '{task_id}' referenced by collection '{collection_id}' "
+                                        f"was not found in yaml files")
+                    else:  # check the task
+                        task = tasks[task_id]
+                        instructions: Dict[str, InstructionArgs] = meta.read_instructions()
+                        instruction_id = task.instruction_id
+                        if instruction_id and not instruction_id in instructions:
+                            raise PathError(f"Instruction '{instruction_id}' referenced by task '{task_id}' "
+                                            f"was not found in the yaml files. "
+                                            f"Task '{task_id}' is referenced by collection '{collection_id}'.")
+                        stimuli: Dict[str, StimulusArgs] = meta.read_stimuli()
```

Comment:
Are we reloading the same mapping in every loop here?

Reply:
Yes.  I didn't think it mattered as it is 'fast enough' for something that gets run only when someone makes a structural change to the data. OTOH, we do need it today if we are deploying the new files to Wang tomorrow morning.

Reply:
We're not really working at a large enough scale for the performance hit to matter, so I think it's fine for now. Just a small comment for if we end up rewriting a bit later.